WcaOnRails/lib/competition_results_validator.rb

```diff
@@ -589,14 +593,22 @@ def check_individual_results(results_by_round_id)
             cumulative_round_ids = cumulative_wcif_round_ids.map do |wcif_id|
               parsed_wcif_id = Round.parse_wcif_id(wcif_id)
               # Get the actual round_id from our expected rounds by id
-              @expected_rounds_by_ids.select do |id, round|
+              actual_round_id = @expected_rounds_by_ids.select do |id, round|
                 round.event.id == parsed_wcif_id[:event_id] && round.number == parsed_wcif_id[:round_number]
-              end.first[0]
-            end
+              end.first
+              unless actual_round_id
+                # FIXME: this needs to be removed when https://github.com/thewca/worldcubeassociation.org/issues/3254 is fixed.
+                @errors[:results] << format(MISSING_CUMULATIVE_ROUND_ID_ERROR, wcif_id: wcif_id, original_round_id: round_id)
+              end
+              actual_round_id&.at(0)
+            end.compact
 
             # Get all solve times for all cumulative rounds for the current person
             all_results_for_cumulative_rounds = cumulative_round_ids.map do |id|
-              results_by_round_id[id].find { |r| r.personId == result.personId }
+              # NOTE: since we proceed with all checks even if some expected rounds
```

Comment:
Should this also get removed when we fix #3254? If so, maybe worth updating this comment to a `FIXME` as well?

Reply:
No: it's a situation that can occur with a "normal" error.
Eg: you plan to have 444bf and 555bf with a cumulative time limit, but you end up having no results for one of the events.
Until you edit the events, you can still end up in this situation.